src/main/java/scratch/UCERF3/inversion/InversionTargetMFDs.java

```diff
@@ -155,7 +155,7 @@ private Precomputed() {
 		public Precomputed(InversionTargetMFDs targetMFDs) {
 			this(targetMFDs.rupSet, targetMFDs.getTotalRegionalMFD(), targetMFDs.getTotalOnFaultSupraSeisMFD(),
 					targetMFDs.getTotalOnFaultSubSeisMFD(), targetMFDs.getTrulyOffFaultMFD(),
-					targetMFDs.getMFD_Constraints(), targetMFDs.getOnFaultSubSeisMFDs());
+					(List<MFD_InversionConstraint>) targetMFDs.getMFD_Constraints(), targetMFDs.getOnFaultSubSeisMFDs());
```

Comment:
In concert with my suggested adapter change, you could make the other `Precomputed` constructor take a `List<? extends MFD_InversionConstraint>` and remove this cast

Reply:
done